src/components/Hosts/HostPopover.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState, useRef, useEffect } from "react";
+import styled from "@emotion/styled";
+import { uiColors } from "@leafygreen-ui/palette";
+import Popover from "@leafygreen-ui/popover";
+import { Button } from "components/Button";
+
+const { gray } = uiColors;
+
+interface Props {
+  buttonText: string;
+  titleText: string;
+  loading: boolean;
+  disabled: boolean;
+  onClick: () => void;
+  "data-cy"?: string;
+}
+
+export const HostPopover: React.FC<Props> = ({
+  buttonText,
+  titleText,
+  loading,
+  disabled = false,
+  onClick,
+  "data-cy": dataCy,
+}) => {
+  const [active, setActive] = useState(false);
+  const buttonRef = useRef(null);
+  const popoverRef = useRef(null);
+
+  // Handle onClickOutside
+  useEffect(() => {
+    if (!active) {
+      return;
+    }
+    const onClickOutside = (event: MouseEvent) => {
+      const stillFocused =
+        popoverRef.current!.contains(event.target as Node) ||
+        buttonRef.current!.contains(event.target as Node);
+      setActive(stillFocused);
+    };
+
+    document.addEventListener("mousedown", onClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", onClickOutside);
+    };
+  }, [buttonRef, popoverRef, active]);
```

Comment:
I was going to use [the hook](https://github.com/evergreen-ci/spruce/blob/92f2dbe3f6c8104e0bc2313b76489e28a6d0d647/src/hooks/useOnClickOutside.ts) but then I found that I needed to use two `refs` simultaneously.

It's because clicking the button when the popover is already open should close the popover, but the `popoverRef` interferes with that behavior. The `popoverRef` will detect that click as an outside click and close the button, so the button click will just open the popover again (i.e. it's not possible to close the popover). The behavior is shown in the video below:

https://user-images.githubusercontent.com/47064971/142654379-8c6fafc2-5f0a-4d23-83f9-b46f770340d1.mov

Reply:
Could we extend that hook to maybe allow passing in many refs?  It would also be useful here. https://github.com/evergreen-ci/spruce/blob/main/src/components/SearchableDropdown/index.tsx#L51-L66

Reply:
I'll look into it 🥸